### PR TITLE
hdl.mem: document ReadPort and WritePort.

### LINKED ISSUE #496: no doc string for `en` in `ReadPort`
typing `help(Memory(width=1,depth=1).read_port())` into the interpreter yields no information about the `en` signal of the read port.

### PR DESCRIPTION
**Pull request by jfng**
Fixes #496.

### CODE REVIEW

File: nmigen/hdl/mem.py [hunk 1]
```diff
@@ -92,6 +118,37 @@ def __getitem__(self, index):
...
+    memory : :class:`Memory`
+    domain : str
+    transparent : bool
+    addr : Signal(range(memory.depth)), in
+        Read address.
+    data : Signal(memory.width), out
+        Read data.
+    en : Signal or Const, in
+        Read enable. If asserted, ``data`` is updated with the word stored at ``addr``. If the
+        port is transparent, it is hardwired to 1 and cannot be driven by user logic.
```
Comment (whitequark) [inline nmigen/hdl/mem.py:146]: It's worth mentioning that this hardwiring behavior is not what we want to do, but in fact the consequence of a Yosys design issue.

File: nmigen/hdl/mem.py [hunk 2]
```diff
@@ -92,6 +118,37 @@ def __getitem__(self, index):
...
 class ReadPort(Elaboratable):
+    """A memory read port.
+
+    Parameters
+    ----------
+    memory : :class:`Memory`
+        Memory associated with the port.
+    domain : str
+        Clock domain. Defaults to ``"sync"``. If set to ``"comb"``, the port is asynchronous.
+        Otherwise, reads have a latency of 1 clock cycle.
```
Comment (whitequark) [inline nmigen/hdl/mem.py:129]: I would rephrase this along the lines of "the read data becomes available on the next clock cycle", since otherwise there's a bit of ambiguity with pipelined reads where that'd be an *extra* clock cycle.

**Review by whitequark — CHANGES_REQUESTED** [review]
Good work! There's a few clarifications I would add.

**Comment by jfng** [comment]
Updated to address previous comments.

**Review by whitequark — APPROVED** [review]
